src/suggestion-box.ts

```diff
@@ -0,0 +1,409 @@
+
+    @property()
+    highlight: string = "";
+
+    @property()
+    input_el: HTMLElement | null = null;
+
+    @property()
+    filtered_list: Array<{ [key: string]: string }> = [];
+
+    @property()
+    prevIndex: number = 0;
+
+    @property()
+    minchar: number = 1;
+
+    @state()
+    show_container = false;
+
+    @state()
+    has_more: boolean = false;
+
+    @state()
+    index: number = -1;
+    @state()
+    show_empty: boolean = true;
+
+    private _outClickListener = this.windowClickEvent.bind(this);
+
+    override connectedCallback() {
+        super.connectedCallback();
+        this.addEventListener("mouseout", this._mouseOutEvent);
+        document.addEventListener("click", this._outClickListener);
+    }
+    override disconnectedCallback() {
+        super.disconnectedCallback();
+        this.removeEventListener("mouseout", this._mouseOutEvent);
+        document.removeEventListener("click", this._outClickListener);
+    }
+    override render() {
+        return html`
+            <div style="position:relative">
+                <div
+                    part="sgn-input"
+                    class="inputbox"
+                    @focusin=${this._inputFocusEvent}
+                    @keydown=${this.keyDownInputEvent}
+                >
+                    <slot
+                        name="sgn-input"
+                        id="sgn-input"
+                        @input=${this._inputEvent}
+                        @click=${this._inputSelectEvent}
+                        ><input type="text" />
+                    </slot>
+                </div>
+                <slot name="sgn-item-template" style="display:none"></slot>
+                <div
+                    id="sgn-container"
+                    part="sgn-container"
+                    style="display:${this.show_container ? "block" : "none"};"
+                >
+                    <div style="display:${this.show_empty ? "block" : "none"}">
+                        <slot id="sgn-empty" name="sgn-empty"></slot>
+                    </div>
+                    <div
+                        style="display:${!this.show_empty ? "block" : "none"};"
+                        @click=${this._saveInputEvent}
+                        @mouseover=${this._mouseOverEvent}
+                    >
+                        <slot id="private-item" name="__private-item"></slot>
+                    </div>
+                </div>
+            </div>
+        `;
+    }
+    private _indexOfTarget(event: any) {
+        let el = event.target;
+        while (el) {
+            if (el.slot == "__private-item") break;
+            el = el.parentElement;
+        }
+        return this.private_items.indexOf(el);
+    }
+    private _saveInputEvent(event: Event) {
+        let indextarget = this._indexOfTarget(event);
+        if (this.result[indextarget]) {
+            (this.input_el as HTMLInputElement).value =
+                this.result[indextarget][this.listkey];
```

Comment:
The result can be an array of strings. Better to have a check if result[indexTarget] is of type string and then take the necessary action.